mediaire_toolbox/transaction_db/migrations.py

```diff
@@ -59,6 +59,84 @@
     ],
     16: [
         "ALTER TABLE transactions ADD COLUMN priority INT DEFAULT 0"
+    ],
+    17: [
+        "CREATE TABLE IF NOT EXISTS sites (id INT PRIMARY KEY, name TEXT)",
+        "INSERT OR IGNORE INTO sites (id, name) VALUES (0, 'default')",
+        # FOREIGN KEY column needs to be created via temporary table due to
+        # sqlite's limited ALTER TABLE support: www.sqlite.org/faq.html#q11
+        ("CREATE TEMPORARY TABLE transactions_backup ("
```

Comment:
I'm not sure we're talking about the same thing here. This is not a backup for the sake of being able to roll back the migration in case of an issue. This is a temporary table to store the data in _during_ migration. sqlite's `ALTER TABLE` does not support adding `FOREIGN KEY`s to exisiting tables.

So in order to create the `FOREIGN KEY` column, I need to delete the whole table, create a new one with the desired column and then restore the data from the old table. Hence a backup table.

Am I misunderstanding something here and do you still want me to get rid of this? I think the migration will fail if we exclude this...

Reply:
Thanks, I understand now. I'm just a bit afraid of such big migrations. I don't know how this will perform with a 3 GB t.db file. Is it really worth it for just having a foreign key constraint? I dont' know. You decide, but then at least we need to know how much time it will take to do run this command on an average SSD disk and a 3GB table. And what would happen if the support person cancels the operation before it being finished (accidentally). In the past we had already a big migration for t.db and such things happened more than not (unfortunately). We should also confirm that before any migration there is an automatic copy of t.db being made (I think Wenxin implemented that already).

Reply:
If you put it like this I actually don't think it's worth it... :sweat_smile: Currently the "sites" table is not used for anything, I just included it because I though it made sense. I'd keep it, but we definitely don't need the foreign key constraint. If we actually need it, we could add it later. Joins should be still possible even without the foreyn key, (maybe less optimized?), if we do need them at some point.

Reply:
@pereferrera I removed the FOREIGN KEY constraint in the latest commit.